CMakeLists.txt

```
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(ClpIrV1Decoder)
+
+# Download and extract Boost
+set(BOOST_VERSION "1.85.0")
+string(REPLACE "." "_" BOOST_ARCHIVE_VERSION_PART ${BOOST_VERSION})
+
+include(ExternalProject)
+ExternalProject_Add(
+        Boost
+        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
+        PREFIX "${CMAKE_BINARY_DIR}/boost"
+        CONFIGURE_COMMAND ""
+        BUILD_COMMAND ""
+        INSTALL_COMMAND ""
+        LOG_DOWNLOAD ON
+)
+ExternalProject_Get_Property(Boost SOURCE_DIR)
+set(BOOST_EXTRACT_DIR "${SOURCE_DIR}")
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++23")
+set(CMAKE_CXX_STANDARD 23)
+
+set(CMAKE_EXECUTABLE_SUFFIX ".js")
+
+add_executable(
+        ClpIrV1Decoder
+
```

See above for the comment to separate `add_executable`.
```suggestion
target_sources(${BIN_NAME} PRIVATE
```

---

After applying all suggestions, let's place this `target_sources` block at the end of the place so the sources are specified after all compilation options and include directory specifications.